mesop/examples/custom_component/custom_component.js

```diff
@@ -0,0 +1,51 @@
+import {
+  LitElement,
+  html,
+  css,
+} from 'https://cdn.jsdelivr.net/gh/lit/dist@3/core/lit-core.min.js';
+
+class FooComponent extends LitElement {
+  static properties = {
+    value: {type: Number},
+    style: {type: String},
+    handlerId: {type: String},
+  };
+
+  constructor() {
+    super();
+    this.value = 0;
+    this.style = '';
+    this.handlerId = '';
+  }
+
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  render() {
+    return html`
+      <div class="container" style="${this.style}">
+        <span>Value: ${this.value}</span>
+        <button id="increment-btn" @click="${this._onIncrement}">
+          Increment
+        </button>
+      </div>
+    `;
+  }
+
+  _onIncrement() {
+    this.dispatchEvent(
+      new CustomEvent('mesop-event', {
```

Comment:
fwiw, I recommend against using CustomEvent these days. It's a vestige of a time when `Event` wasn't subclassable. You can vend a MesopEvent class like:

```ts
export class MesopEvent<T> {
  payload: T;
  handlerId: string;
    
  constructor(payload: T, handlerId: string) {
    super('mesop-event', {bubbles: true});
    this.payload = payload;
    this.handlerId = handlerId;
  }
}
```

The advantage here is that you ensure consistent event dispatching with that constructor:

```ts
  _onIncrement() {
    this.dispatchEvent(new MesopEvent({value: this.value + 2}, this.handlerId);
  }
```

Of course, this requires that you vend a JS library somewhere to define that MesopEvent - not sure if you have one already.

Reply:
Thanks for the suggestion.

Right now, Mesop doesn't have a published JS library, I'm not sure if we should:
a) publish a library on npm - but then this would require Mesop developers to install npm & build their JS binary. I think this would be a pretty big barrier for many (but not all) Mesop developers given their python-centric backgrounds.
b) publish a library on a CDN and have developers import it directly there - similar to [Lit bundles](https://lit.dev/docs/getting-started/#use-bundles)
c) provide MesopEvent on the window, e.g. window["MesopEvent"]

We could also support some combination (eventually).